src/ast.ts

```diff
@@ -1119,6 +1144,11 @@ export class NamedTypeNode extends TypeNode {
   name: TypeName;
   /** Type argument references. */
   typeArguments: TypeNode[] | null;
+  equals(node: NamedTypeNode): bool {
```

Comment:
Considering that type names can be different depending on context, this seems odd. For example, the class B in

```ts
namespace A {
  class B {}
}
```

is `A.B` from the outside but can also be `B` from the inside. One can also import `A as C` in a different file, making that `C.B`.

Reply:
Yeah I added this to compare signatures, so I just need to resolve the functions with the type information of the generic method/bound class and check if the signatures are equal.

Reply:
Wouldn't that still apply if the signature looks like `(a: A.B) => A.B`?

Reply:
I'm not sure I follow in this situation.  Could you provide an example with an interface?

Reply:
Not sure I am following either, but for interfaces maybe something like this?

```ts
namespace A {
  export interface B { foo: i32; }
  interface D extends B {} // == A.B
}
interface C extends A.B {}
```

Reply:
I think that this is not as issue if we resolve the types to their respective interfaces and check when used if the underlying class implements it.  Before I was comparing function prototypes, but if I use new function resolving logic then it won't be a syntactic comparison, so names shouldn't matter.  Does that make sense?